src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -573,16 +573,22 @@ private static Long findSchema(Connection connection, Position targetPosition, L
 			// Only consider binlog positions before the target position on the current server.
 			// Within those, sort for the latest binlog file, then the latest binlog position.
 			PreparedStatement s = connection.prepareStatement(
-				"SELECT id from `schemas` "
-				+ "WHERE deleted = 0 "
-				+ "AND last_heartbeat_read <= ? AND ((binlog_file < ?) OR (binlog_file = ? and binlog_position < ?)) AND server_id = ? "
-				+ "ORDER BY last_heartbeat_read DESC, binlog_file DESC, binlog_position DESC limit 1");
+				"SELECT id from `schemas` " +
+				"WHERE deleted = 0 AND " +
+				"last_heartbeat_read <= ? AND " +
+			    "((binlog_file < ?) OR " +
+				"(binlog_file = ? AND binlog_position < ?) OR " +
```

Comment:
Understood and nice fix, may I know where is the code to write `deltas` to be `[]` when master is failover ?

Reply:
Also, is there any other cases that deltas is written to `[]` but it is actually not a failover ? 
I could be wrong but I feel like there are several schema deltas values have special meanings (little confusing and little hard to maintain):

- Empty -> Beginning
- [{...}] -> Real delta and schema
- [] -> Master failover

Reply:
It's in `MysqlSchemaStore` clone method where we specify an empty list for `deltas`: https://github.com/zendesk/maxwell/blob/master/src/main/java/com/zendesk/maxwell/schema/MysqlSchemaStore.java#L122-L134

We only write `[]` as deltas for master failover case AFAIK.